PRESIDE-1056 Allow for sites to run in subfolder of root

patmoody opened the pull request:
Application config example
settings.static.rootUrl = "/cms_p";
settings.preside_admin_base_path = "cms_p/";

Application.cfc
super.setupApplication( id = "mysite", appBasePath = "/cms_p" );

Site path in DB
update psys_site set path = '/cms_p';

Commit:
PRESIDE-1056 Allow for sites to run in subfolder of root

Commit:
PRESIDE-1056 Allow for sites to run in subfolder of root

patmoody commented:
This is a start but still needs some work! Failing in a standard install with site in root location. Looks like some Asset management pathing issues too with site outside root.

Commit:
feature-PRESIDECMS-1056 Allow for site to run in subfolder of root

Commit:
feature-PRESIDECMS-1056 Allow for site to run in subfolder of root

patmoody commented:
Fixed for standard installation. Asset management pathing still a problem for non standard (Site in subfolder)

Commit:
PRESIDE-1056 Allow for sites to run in subfolder of root - Update for assets.

Commit:
PRESIDE-1056 Allow for sites to run in subfolder of root - Update for assets.

patmoody commented:
Fixed for assets too. 
@DominicWatson 
Ready for review.

Commit:
Merge branch 'release-10.9.0' into feature-PRESIDECMS-1056_Allow_for_sites_to_run_in_subfolder_of_root

Commit:
PRESIDE-1056 PRESIDE-1056 Allow for sites to run in subfolder of root - Deal with assets

Commit:
PRESIDE-1056 PRESIDE-1056 Allow for sites to run in subfolder of root - Deal with assets

patmoody commented on system/services/assetManager/AssetManagerService.cfc:
```diff
@@ -1870,6 +1874,14 @@ component displayName="AssetManager Service" {
 		_setTypes( types );
 	}
 
+	private string function _getAssetPath( string appBasePath="/asset/" ) {
```
This of course should be _getAssetPath() - Will await other instructions before committing again

DominicWatson commented:
Fab, thanks for this @patmoody. I'll take a good look shortly.

Commit:
Merge branch 'release-10.9.0' into feature-PRESIDECMS-1056_Allow_for_sites_to_run_in_subfolder_of_root

Commit:
PRESIDECMS-1056 Allow for sites to run in subfolder of root - based on site.path

Commit:
PRESIDECMS-1056 Allow for sites to run in subfolder of root - based on site.path

patmoody commented on system/Bootstrap.cfc:
```diff
@@ -546,6 +551,32 @@ component {
+		if ( Len( arguments.appBasePath ) ) {
+			appBasePath = arguments.appBasePath;
+		} else {
+			var cbController = _getColdboxController();
```
@DominicWatson 
Trying to get this to work based on site.path if no configuration options are passed through / set in the applications config.
Thought I had this working locally but just realised I was passing the site path in the applications app cfc file.
cbController is returning as null when cfg options are removed.

patmoody commented:
Latest commit was trying to avoid settings in the applications config and application.cfc and basing the root path on the site.path. - bootstrap.cfc - _getAppBasePath
cbController = _getColdboxController() 
returns null

DominicWatson commented:
Hey Pat, apologies I've not managed to look at this yet. I'll defo look at this before we release 10.9. Thanks for keeping going with it

DominicWatson committed:
PRESIDECMS-1022 change UI to improve UX of tracking background tasks

Make the progress bar transition from animated + striped to a solid colour when the task is no longer running

DominicWatson committed:
PRESIDECMS-1022 change export file download logic to improve UX

Expire the download file in 10m, rather than immediately. Will cause less problems with users encountering 404s when hitting refresh, etc.

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-1022_ad-hoc-taskmanager-tasks' into release-10.9.0

patmoody commented:
No probs. My problem now is checking the site.path programmatically from the bootstrap.cfc ^^

DominicWatson committed:
PRESIDECMS-906 fix extension load order through the presidecms:directories injection

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-906-extensions-in-cb-modules' into release-10.9.0

DominicWatson committed:
[twgit] Init feature 'feature-PRESIDECMS-1077_sql-script-generator-fix-now-refs'.

DominicWatson committed:
PRESIDECMS-1077 use calculated Now() function in schema upgrade script generation

Fixes issue where script will not run on non-MySQL databases

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-1077_sql-script-generator-fix-now-refs' into release-10.9.0

jjannek committed:
added MsSql2012 compatible concatenation function to DB adapter

idg-germany committed:
Fix for labelRenderer search & MSSQL select

DominicWatson committed:
PRESIDECMS-906 fix eternal loop with some extension dependencies

DominicWatson committed:
[twgit] Init feature 'feature-PRESIDECMS-1078_mssql_2012_concat'.

DominicWatson committed:
Merge branch 'mssql2012_fix_concatenation' of https://github.com/jjannek/Preside-CMS into feature-PRESIDECMS-1078_mssql_2012_concat

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-1078_mssql_2012_concat' into release-10.9.0

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-906-extensions-in-cb-modules' into release-10.9.0

DominicWatson committed:
[twgit] Init feature 'feature-PRESIDECMS-1079_objectpicker_labelfield'.

DominicWatson committed:
Merge branch 'release-10.9.0' into feature-PRESIDECMS-1079_objectpicker_labelfield

DominicWatson committed:
Merge branch 'PRESIDECMS-labelRenderer_mssql_fix' of https://github.com/thomasrotter/Preside-CMS into feature-PRESIDECMS-1079_objectpicker_labelfield

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-1079_objectpicker_labelfield' into release-10.9.0

sebduggan committed:
PRESIDECMS-1024 Resend emails using original args

sebduggan committed:
PRESIDECMS-1024 Add rebuildArgsForResend method to system email templates

sebduggan committed:
PRESIDECMS-1024 Log content of sent emails

DominicWatson committed:
[twgit] Init hotfix 'hotfix-10.8.60'.

DominicWatson committed:
PRESIDECMS-1076 remove group by on query that does not need it

This fixes issues on MSSQL where records not getting selected

DominicWatson committed:
PRESIDECMS-1088 use new way of getting record count, avoiding unnecessary order by clause in grouped query

DominicWatson committed:
Merge branch 'hotfix-10.8.60' into stable

DominicWatson committed:
Merge tag 'v10.8.60' into release-10.9.0

[twgit] Hotfix finish: hotfix-10.8.60

DominicWatson committed:
PRESIDECMS-931 fix bad reference to variable

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-931_rest-api-manager' into release-10.9.0

DominicWatson committed:
[twgit] Init feature 'feature-PRESIDECMS-1089_many-to-one-field-datmanager-sort-bug'.

DominicWatson committed:
PRESIDECMS-1089 change logic to ensure property name is used instead of object name in sort field definition

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-1089_many-to-one-field-datmanager-sort-bug' into release-10.9.0

sebduggan committed:
PRESIDECMS-1024 Mechanism for resending original email content

sebduggan committed:
PRESIDECMS-1024 Email rebuild method for website password reset

sebduggan committed:
PRESIDECMS-1024 Add expiry to saved email content

sebduggan committed:
PRESIDECMS-1024 Add permissioning to email resend

sebduggan committed:
PRESIDECMS-1024 Scheduled task to delete expired email content

sebduggan committed:
PRESIDECMS-1024 PRESIDECMS-1090 Make task manager respect feature flags

sebduggan committed:
PRESIDECMS-1024 Fix email sending tests to match recent changes

sebduggan committed:
PRESIDECMS-1024 Use feature for email resend. Disabled by default.

sebduggan committed:
PRESIDECMS-1024 Email template configure whether to save content or not

sebduggan committed:
PRESIDECMS-1024 Add save content and expiry settings to newsletters

sebduggan committed:
PRESIDECMS-1024 Change label on "rebuild and resend" button

sebduggan committed:
PRESIDECMS-1024 HTML email log view for customTemplates

sebduggan committed:
PRESIDECMS-1024 Chnage resend feature name to "emailCenterResend"

sebduggan committed:
PRESIDECMS-1024 Fix EmailTemplateService test

sebduggan committed:
PRESIDECMS-1024 Email content expiry of 0 should not log content

sebduggan committed:
PRESIDECMS-1024 Documentation for email resending

sebduggan committed:
PRESIDECMS-1024 Rename resendEmail to resendEmailAction

sebduggan committed:
PRESIDECMS-1024 Add audit trail for email resending

DominicWatson committed:
[twgit] Init feature 'feature-PRESIDECMS-1024_email-resending'.

DominicWatson committed:
Merge branch 'release-10.9.0' into feature-PRESIDECMS-1024_email-resending

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-1024_resend-emails' of https://github.com/sebduggan/Preside-CMS into feature-PRESIDECMS-1024_email-resending

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-1024_email-resending' into release-10.9.0

patmoody committed:
feature-PRESIDECMS-1056 Merge of latest 10.9.0 + Bootstrap typo and config uploads_directory

patmoody committed:
feature-PRESIDECMS-1056 Merge of latest 10.9.0 + Bootstrap typo and config uploads_directory

jjannek committed:
escaping db columns that are reserved words in ms sql

john5onCheng committed:
PRESIDECMS-1091
Corrected the form builder parameter values

DominicWatson committed:
[twgit] Init hotfix 'hotfix-10.8.61'.

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-1091_formbuilder_form' of https://github.com/john5onCheng/Preside-CMS into hotfix-10.8.61

DominicWatson committed:
Merge branch 'mssql_escape_reserved_word_fix' of https://github.com/jjannek/Preside-CMS into hotfix-10.8.61

DominicWatson committed:
Merge branch 'hotfix-10.8.61' into stable

DominicWatson committed:
Merge tag 'v10.8.61' into release-10.9.0

[twgit] Hotfix finish: hotfix-10.8.61

john5onCheng committed:
PRESIDE-1098
Corrected the default returned value when session is not enabled

DominicWatson committed:
[twgit] Init hotfix 'hotfix-10.8.62'.

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-1098_session_storage_default_return_value' of https://github.com/john5onCheng/Preside-CMS into hotfix-10.8.62

DominicWatson committed:
PRESIDECMS-1099 add site template to merged form names so that merged forms are distinct per site template

DominicWatson committed:
Fix misisng semi-colon

DominicWatson committed:
Merge branch 'hotfix-10.8.62' into stable

DominicWatson committed:
Merge branch 'stable' into release-10.9.0

DominicWatson committed:
[twgit] Init feature 'feature-PRESIDECMS-1072_formula-fields-in-datamanager-search'.

DominicWatson committed:
PRESIDECMS-1072 expand formula fields in datamanager grid listings / search

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-1072_formula-fields-in-datamanager-search' into release-10.9.0

john5onCheng committed:
PRESIDECMS-1101
Correction on variable name

john5onCheng committed:
PRESIDECMS-1102
Corrected min length value

jjannek committed:
fixed one time scheduled email template sending (query to find relevant templates did not look after null values for flag indicating that schedule was sent)

jjannek committed:
fixed unit test

DominicWatson committed:
[twgit] Init feature 'feature-PRESIDECMS-1103_i18n-debug'.

DominicWatson committed:
PRESIDECMS-1103 add feature to allow developers to debug i18n resource URIs

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-1103_i18n-debug' into release-10.9.0

DominicWatson committed:
[twgit] Init feature 'feature-translations'.

DominicWatson committed:
Latest 10.9 translations

DominicWatson committed:
Merge branch 'feature-translations' into release-10.9.0

DominicWatson committed:
[twgit] Init hotfix 'hotfix-10.8.63'.

DominicWatson committed:
Add latest translations to 10.8

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-1102_incorrect_minlength' of https://github.com/john5onCheng/Preside-CMS into hotfix-10.8.63

DominicWatson committed:
Merge branch 'fix_email_centre_one_time_scheduling' of https://github.com/jjannek/Preside-CMS into hotfix-10.8.63

DominicWatson committed:
Merge branch 'hotfix-10.8.63' into stable

DominicWatson committed:
Merge tag 'v10.8.63' into release-10.9.0

[twgit] Hotfix finish: hotfix-10.8.63

DominicWatson committed:
[twgit] Init feature 'feature-PRESIDECMS-1101_assertversionupdate_typo'.

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-1101_10.9_error_typo_variable_name' of https://github.com/john5onCheng/Preside-CMS into feature-PRESIDECMS-1101_assertversionupdate_typo

DominicWatson committed:
Merge branch 'feature-PRESIDECMS-1101_assertversionupdate_typo' into release-10.9.0

patmoody committed:
feature-PRESIDECMS-1056 settings.preside_admin_base_path

patmoody committed:
Merge remote-tracking branch 'upstream/feature-PRESIDECMS-1056_Allow_for_sites_to_run_in_subfolder_of_root' into feature-PRESIDECMS-1056_Allow_for_sites_to_run_in_subfolder_of_root

patmoody committed:
feature-PRESIDECMS-1056 appBasePath - require help with Bootstrap.cfc _getAppBasePath

DominicWatson commented:
Right, finally getting round to looking into this and there appears to be an issue with the pull request. The branch you target the pull request back into should be the branch you have based your feature branch from. e.g. if you did:

```git checkout -b mynewfeature stable```

or, you had `stable` checked out and you did:

```git checkout -b mynewfeature```

...then your pull request should be aimed at `stable`. There are a lot of changes here unrelated to your work I presume (160+ files changed). 

Could you take a look and resubmit by closing this and submitting a new pull request? See our contrib guidelines for more details: https://beta-docs.preside.org/contribguides/submittingchanges.html
